app/models/legacy/loan_response_set.rb

```diff
@@ -33,49 +33,58 @@ def self.migrate(response_set_id)
       puts "responses count: #{responses.count}"
       responses.each do |response|
         puts "response id: #{response.id} - question id: #{response.question_id}"
         field = CustomField.find_by(id: response.question_id)
         if field
-          puts "question_id: #{response.question_id} - set: #{field.custom_field_set.internal_name}, attr: #{attribute_name}"
-          # model = new_loan.fetch_belongs_to_custom(attribute_name, field_set_name: field.custom_field_set.internal_name,
-          #                                          owner: new_loan.organization, autocreate: true)
-          model = new_loan.fetch_has_one_custom(field.custom_field_set.internal_name, autocreate: true)
-          # note, could be optimized by building entire json blob and storing as a single operations, but this seems fast enough
-          puts "update: #{field.id} -> #{response.value_hash}"
-          model.update_custom_value(field.id, response.value_hash)
+          puts "question_id: #{response.question_id} - set: #{field.custom_field_set.internal_name}"
+          model = models[field.custom_field_set.internal_name]
+          unless model
+            model = new_loan.fetch_has_one_custom(field.custom_field_set.internal_name, autocreate: true)
+            models[field.custom_field_set.internal_name] = model
+          end
+          # puts "update: #{field.id} -> #{response.value_hash}"
+
+          value_hash = response.value_hash
+          embeddable_media_id = value_hash.delete(:embeddable_media_id)
+          if embeddable_media_id
+            puts "updating owner of embeddable media: #{embeddable_media_id} to cvs #{model.id}"
+            embeddable_media = EmbeddableMedia.find_safe(embeddable_media_id)
+            if embeddable_media
+              embeddable_media.update!(owner: model, owner_attribute: field.id)
+            else
+              puts "**** warning, dangling embeddable media ref: #{embeddable_media_id} by cvs: #{model.id}"
+            end
+          end
+
+          model.custom_data[field.id.to_s] = value_hash
         else
           puts "WARNING - custom field not found for id: #{response.question_id}"
         end
       end
 
-      # LoanResponseSet.where("ResponseSetID = ? and ResponseSetID <> LoanID", response_set_id).pluck('LoanID').each do |linked_loan_id|
-      #   puts "related loan: #{linked_loan_id}"
-      #   link_response(new_loan, linked_loan_id)
-      # end
-
+      models.values.each do |m|
+        # Some migration data exists with invalid numeric data.  Need to disable validation here
+        # so that the full set of migrated data will be persisted.  A warning will be displayed
+        # when a record with invalid data is first loaded (i.e. loan 1043 post analysis).
+        # The simple form automatic numeric field handler will automatically strip the invalid
+        # numeric data so that the record can still be simply re-saved.
+        m.save!(:validate => false)
```

Comment:
Yes ! just means raise error on validation failure.

On Thu, Aug 4, 2016 at 10:48 AM, Joseph Elliott <notifications@github.com>
wrote:

> In app/models/legacy/loan_response_set.rb
> <https://github.com/sassafrastech/madeline_system/pull/117#discussion_r73536485>
> :
>
> >          else
> >            puts "WARNING - custom field not found for id: #{response.question_id}"
> >          end
> >        end
> >
> > -      # LoanResponseSet.where("ResponseSetID = ? and ResponseSetID <> LoanID", response_set_id).pluck('LoanID').each do |linked_loan_id|
> > -      #   puts "related loan: #{linked_loan_id}"
> > -      #   link_response(new_loan, linked_loan_id)
> > -      # end
> > -
> > +      models.values.each do |m|
> > +        # Some migration data exists with invalid numeric data.  Need to disable validation here
> > +        # so that the full set of migrated data will be persisted.  A warning will be displayed
> > +        # when a record with invalid data is first loaded (i.e. loan 1043 post analysis).
> > +        # The simple form automatic numeric field handler will automatically strip the invalid
> > +        # numeric data so that the record can still be simply re-saved.
> > +        m.save!(:validate => false)
>
> Is it not possible for the save to fail for other reasons, perhaps db
> constraint failure? Or are the '!' generated failures strictly tied to
> validation?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/117/files/c63dba4d1e1ea6309da30d55ac06c390e51c7ac3#r73536485>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq3cLwGLuLjDIoCLsjyMFjELd3yhr5dks5qcfu6gaJpZM4JSkza>
> .
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org